File: client/src/services/healthCenterService.ts
import api from './api';

export interface HealthCenter {
  id: string;
  name: string;
  type: string;
  address?: string;
  city?: string;
  country?: string;
  latitude?: number;
  longitude?: number;
  capacity?: any;
  resources?: any[];
  specialties?: string[];
  operatingHours?: any;
  contactInfo?: any;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface HealthCenterCreate {
  name: string;
  type: string;
  address?: string;
  city?: string;
  country?: string;
  latitude?: number;
  longitude?: number;
  capacity?: any;
  resources?: any[];
  specialties?: string[];
  operatingHours?: any;
  contactInfo?: any;
}

export interface HealthCenterUpdate {
  name?: string;
  type?: string;
  address?: string;
  city?: string;
  country?: string;
  latitude?: number;
  longitude?: number;
  capacity?: any;
  resources?: any[];
  specialties?: string[];
  operatingHours?: any;
  contactInfo?: any;
}

export const healthCenterService = {
  // Obtener todos los centros de salud
  getAll: async (params?: { skip?: number; limit?: number; search?: string; center_type?: string }) => {
    const response = await api.get('/api/v1/config/centers', { params });
    return response.data;
  },

  // Obtener centro por ID
  getById: async (id: string) => {
    const response = await api.get(`/api/v1/config/centers/${id}`);
    return response.data;
  },

  // Crear nuevo centro
  create: async (center: HealthCenterCreate) => {
    const response = await api.post('/api/v1/config/centers', center);
    return response.data;
  },

  // Actualizar centro
  update: async (id: string, center: HealthCenterUpdate) => {
    const response = await api.put(`/api/v1/config/centers/${id}`, center);
    return response.data;
  },

  // Eliminar centro
  delete: async (id: string) => {
    const response = await api.delete(`/api/v1/config/centers/${id}`);
    return response.data;
  },

  // Obtener especialidades de un centro
  getSpecialties: async (id: string) => {
    const response = await api.get(`/api/v1/config/centers/${id}/specialties`);
    return response.data;
  },

  // Agregar especialidad a un centro
  addSpecialty: async (id: string, specialtyId: string) => {
    const response = await api.post(`/api/v1/config/centers/${id}/specialties`, { specialtyId });
    return response.data;
  },

  // Remover especialidad de un centro
  removeSpecialty: async (id: string, specialtyId: string) => {
    const response = await api.delete(`/api/v1/config/centers/${id}/specialties/${specialtyId}`);
    return response.data;
  },
};

File: backend/app/services/health_center_service.py
"""
Servicio para gestión de centros de salud
"""

from sqlalchemy.orm import Session
from typing import List, Optional
import uuid
from datetime import datetime

from app.schemas.health_center import HealthCenterCreate, HealthCenterUpdate, HealthCenterResponse

class HealthCenterService:
    """Servicio para operaciones con centros de salud"""
    
    @staticmethod
    async def get_all(
        db: Session,
        skip: int = 0,
        limit: int = 100,
        search: Optional[str] = None,
        center_type: Optional[str] = None
    ) -> List[HealthCenterResponse]:
        """Obtener todos los centros de salud"""
        # Placeholder - implementar con base de datos real
        return []
    
    @staticmethod
    async def get_by_id(db: Session, center_id: str) -> Optional[HealthCenterResponse]:
        """Obtener centro por ID"""
        # Placeholder - implementar con base de datos real
        return None
    
    @staticmethod
    async def create(db: Session, center: HealthCenterCreate) -> HealthCenterResponse:
        """Crear nuevo centro"""
        # Placeholder - implementar con base de datos real
        center_id = str(uuid.uuid4())
        return HealthCenterResponse(
            id=center_id,
            name=center.name,
            type=center.type,
            isActive=True,
            createdAt=datetime.utcnow(),
            updatedAt=datetime.utcnow()
        )
    
    @staticmethod
    async def update(
        db: Session,
        center_id: str,
        center: HealthCenterUpdate
    ) -> Optional[HealthCenterResponse]:
        """Actualizar centro"""
        # Placeholder - implementar con base de datos real
        return None
    
    @staticmethod
    async def delete(db: Session, center_id: str) -> bool:
        """Eliminar centro"""
        # Placeholder - implementar con base de datos real
        return False





